third_party/lua-cjson/lua_cjson.c

```diff
@@ -423,10 +439,22 @@ static int json_encode(lua_State *l) {
     struct luaL_serializer *cfg = luaL_checkserializer(l);
 
     if (lua_gettop(l) == 2) {
-        struct luaL_serializer user_cfg = *cfg;
-        luaL_serializer_parse_options(l, &user_cfg);
+        /*
+         * user_cfg is per-call local version of serializer instance
+         * options: it is used if a user passes custom options to
+         * :encode() method within a separate argument. In this case
+         * it is required to avoid modifying options of the instance.
+         * Life span of user_cfg is restricted by the scope of
+         * :encode() so it is ok to push it onto the Lua stack and
+         * pop it right away. This way we avoid memory leak, as it will
```

Comment:
> ok to push it onto the Lua stack and pop it right away.

We don't do it now (and it is not right).

Reply:
Thanks, fixed (in both places).